foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/mappings/OneToManyMapping.java

```diff
@@ -982,19 +1045,56 @@ public void postInsert(WriteObjectQuery query) throws DatabaseException, Optimis
-                    updateTargetRowPostInsertSource(query);
+                if ((requiresDataModificationEvents() || containerPolicy.shouldUpdateForeignKeysPostInsert())) {
+                    if (shouldDeferInsert()) {
+                        // Hey I might actually want to use an inner class here... ok array for now.
+                        Object[] event = new Object[2];
+                        event[0] = PostInsert;
+                        event[1] = query;
+                        query.getSession().getCommitManager().addDataModificationEvent(this, event);
+                    } else {
+                        ContainerPolicy cp = getContainerPolicy();
+                        Object objects = getRealCollectionAttributeValueFromObject(query.getObject(), query.getSession());
+                        if (cp.isEmpty(objects)) {
+                            return;
+                        }
+
+                        prepareTranslationRow(query.getTranslationRow(), query.getObject(), query.getDescriptor(), query.getSession());
+
+                        AbstractRecord keyRow = buildKeyRowForTargetUpdate(query);
+
+                        // Extract target field and its value. Construct insert
+                        // statement and execute it
+                        int size = targetPrimaryKeyFields.size();
+                        ClassDescriptor referenceDesc = getReferenceDescriptor();
+                        AbstractSession session = query.getSession();
+                        int objectIndex = 0;
+                        for (Object iter = cp.iteratorFor(objects); cp.hasNext(iter);) {
+                            AbstractRecord row = new DatabaseRecord();
+                            row.mergeFrom(keyRow);
+                            Object wrappedObject = cp.nextEntry(iter, query.getSession());
+                            Object object = cp.unwrapIteratorResult(wrappedObject);
+                            AbstractRecord databaseRow = referenceDesc.getObjectBuilder().buildRow(row, object, session, WriteType.INSERT);
+                            ContainerPolicy.copyMapDataToRow(cp.getKeyMappingDataForWriteQuery(wrappedObject, session), databaseRow);
+                            if (listOrderField != null) {
+                                databaseRow.put(listOrderField, objectIndex++);
+                            }
+                            InsertObjectQuery insertQuery = getInsertObjectQuery(session, referenceDesc);
+                            insertQuery.setObject(object);
+                            insertQuery.setCascadePolicy(query.getCascadePolicy());
+                            insertQuery.setTranslationRow(databaseRow);
+                            insertQuery.setModifyRow(databaseRow);
+                            insertQuery.setIsPrepared(false);
+                            query.getSession().executeQuery(insertQuery);
+                        }
+                    }
+                } else {
```

Comment:
wait, did this get moved on purpose? This if-else is different now.

Reply:
true, if you check full method, it starts with `if (isReadOnly()) { return;}`[see](https://github.com/eclipse-ee4j/eclipselink/blob/8704f7fc17699bb6b02a5381e101a7a9f377fbc4/foundation/org.eclipse.persistence.core/src/org/eclipse/persistence/mappings/OneToManyMapping.java#L1015), so later call to `!isReadOnly()` should be always evaluated to `true`. Do you think there exists a case where this would be evaluated to `false`?

Reply:
Oh, I don't mean the `isReadOnly()` check. I think that should be fine since the method is prefaced with:

``` java
        if (isReadOnly()) {
            return;
        }
```
I mean the different if-else structure.

Reply:
ouch, fixed. Thanks!